tests/lib/script-setup-vars.js

```diff
@@ -0,0 +1,384 @@
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const eslint = require('eslint')
+const rules = new eslint.Linter().getRules()
+const ruleNoUnusedVars = rules.get('no-unused-vars')
+const ruleNoUndef = rules.get('no-undef')
+
+const RuleTester = eslint.RuleTester
+const ruleTester = new RuleTester({
+  parser: require.resolve('vue-eslint-parser'),
+  parserOptions: {
+    ecmaVersion: 6,
+    sourceType: 'module'
+  }
+})
+
+// ------------------------------------------------------------------------------
+// Tests
+// ------------------------------------------------------------------------------
+
+describe('vue-eslint-parser should to properly mark the variables used in the template', () => {
+  ruleTester.run('no-unused-vars', ruleNoUnusedVars, {
```

Comment:
Doesn't it make more sense to put the tests into the respective rule tests?

Reply:
These test that the ESLint core rules work as expected, so I don't think it makes sense to put them in the same place as the plugin rules.

Reply:
Ahh, yeah you're right.